### DISCUSSION TITLE
Is There a Way to Avoid Little Blank Areas?

### DISCUSSION THREAD

**Discussion by ghost** [starship/starship#5536] [Q&A]
Have a look at the Tokyo Night example used in the Starship Docs:

![image](https://github.com/starship/starship/assets/95144826/a3d6a66d-a22d-4e3a-9126-16a1c5441a3d)

Notice how when the Git branch is not present, there is a small blank region between the directory and the time with the same background color as Git branch. I find this undesirable, I would like the region to totally disappear when there is no Git branch. In the Tokyo Night example the  character after the directory name is hardcoded, which makes this impossible. What is needed is a region that always has the same  color, but whose background changes dynamically based on whether the next item (in this case the Git branch) is present or not. So when there is no Git branch you have blue  with background color 1, and when there is a Git branch you have blue  with background color 2. I am unsure how to achieve this or if it is possible, so I have settled on this structure to simply avoid the problem:

![image](https://github.com/starship/starship/assets/95144826/1f45827c-a76f-41ef-bd07-3caf6a82d0b4)

As you can see in this case NO yellow is present when there is no Git branch, however it comes at the cost of a less minimal prompt. I would like to make the prompt more minimal, but I need to solve the dynamic background color problem to achieve this. if you know how to do this please tell me, and if you know that it is impossible please also tell me because then i could put in a feature request. Please feel free to ask me any clarifying quesions.

Thank you very much.

**Comment by davidkna** [comment]
As far as I know, this requires something like #5476 to achieve.

**Comment by mickimnet** [comment]
To extend on what _davidkna_ already wrote, **currently**, if you want a _Powerline_ like style in your prompt, you have two options:
1. always show the arrow for a module, even if the module is empty (like the _Tokoyo Night Preset_)
2. add additional space, like the pill-like prompt you posted

I did the later in my own _Starship_ configuration:
<img width="658" alt="image" src="https://github.com/starship/starship/assets/2911922/bebd40e8-4309-4f97-a519-b33d18b22dc1">

**Reply by ghost** [reply]
Thank you both for the comments. 

davidkna - I would love to see that implemented. Seems like what I want. I have subscribed to the issue.

mickimnet - Yes this is what I had assumed is the case, but I did not feel confident about it because I do not know the software well. So maybe there was a way to get the behaviour I want, that I was simply ignorant of. Hearing others affirm my suspicion raises my credence that it is true. I originally had the powerline style but did not like the empty arrows, so I switched to spaces. I think the empty arrows look nice (you have a nice config for instance), just not the style I am going for.

**Reply by mickimnet** [reply]
You could change the colors of my config easily, as I use `[palettes]`, if that would help you out, in the mean time.

**Reply by ghost** [reply]
Sorry, I missed this reply. Thank you.
